### PR TITLE
Dev modified predator prey

### PR DESCRIPTION
**Pull request by yifan123**
Add a modified predator-prey environment as in [QTRAN](https://arxiv.org/abs/1905.05408) paper to evaluate the superiority of QTRAN than QMIX. The predators get a team reward of `10`, if two or more catch a prey at the same time, but they are given negative reward `−P`, when only one predator catches the prey. This setting requires a higher degree of cooperation.

### CODE REVIEW

File: dizoo/multiagent_particle/envs/multiagent/scenarios/simple_tagv1.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    def adversaries(self, world):
+        return [agent for agent in world.agents if agent.adversary]
+
+    def reward(self, agent, world):
+        # Agents are rewarded based on minimum agent distance to each landmark
+        main_reward = self.adversary_reward(agent, world) if agent.adversary else self.agent_reward(agent, world)
+        return main_reward
+
+    def agent_reward(self, agent, world):
+        # Agents are negatively rewarded if caught by adversaries
```
Comment (PaParaZz1) [inline dizoo/multiagent_particle/envs/multiagent/scenarios/simple_tagv1.py:97]: remove these commented code

File: dizoo/multiagent_particle/envs/particle_env.py [hunk 2]
```diff
@@ -207,7 +374,8 @@ def step(self, action: list) -> BaseEnvTimestep:
         rew_n = rew_n / (self._max_step * self._n_agent)
         self._sum_reward += rew_n
         occupied_landmarks = info['n'][0][3]
-        if self._step_count >= self._max_step or occupied_landmarks >= self._n_agent or occupied_landmarks >= self._num_landmarks:
+        if self._step_count >= self._max_step or occupied_landmarks >= self._n_agent \
+                or occupied_landmarks >= self._num_landmarks:
```
Comment (PaParaZz1) [inline dizoo/multiagent_particle/envs/particle_env.py:378]: add a new doc file named `modified_predator_result.md` (located in `dizoo/multiagent_particle/`), introduce this modified env and show the difference between QMIX and QTran

**Review by PaParaZz1 — APPROVED** [review]
This PR can merged after fixing 3 tiny comments.

File: ding/rl_utils/acer.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from typing import Tuple, List
 from collections import namedtuple
 import torch
-from torch.functional import Tensor
+# from torch.functional import Tensor
```
[the remark this answers is not in the document]
Comment (yifan123) [reply ding/rl_utils/acer.py:4]: this line code requires torch >1.3.1, so i commented this line
image
Version requirements is not right for `from torch.functional import Tensor`
